From 5658a36c6d0dab3c9bcedb936f45f16c3ebc3099 Mon Sep 17 00:00:00 2001
From: Stivvo <stivvo01@gmail.com>
Date: Fri, 28 Aug 2020 16:52:28 +0200
Subject: [PATCH 1/2] Restore tiling

On the main version, when explicitly setting the tiling layout (mod+t)
floating windows don't become tiling. This patch changes this
behaviour.

This allows to switch more easily between tiling and floating, without
manually setting each window as tiling (mod+shift+t) once back in tiling
mode
---
 dwl.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/dwl.c b/dwl.c
index 730e46a1..01728f5a 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1392,8 +1392,15 @@ setlayout(const Arg *arg)
 		return;
 	if (!arg || !arg->v || arg->v != selmon->lt[selmon->sellt])
 		selmon->sellt ^= 1;
-	if (arg && arg->v)
+	if (arg && arg->v) {
+		if (arg->v) { // setting a layout != floating
+			Client *c;
+			wl_list_for_each(c, &clients, link) {
+				setfloating(c, 0);
+			}
+		}
 		selmon->lt[selmon->sellt] = (Layout *)arg->v;
+	}
 	strncpy(selmon->ltsymbol, selmon->lt[selmon->sellt]->symbol, LENGTH(selmon->ltsymbol));
 	arrange(selmon);
	printstatus();
 }

From f4b12e6937dbb849427379974d08f7f0bafec951 Mon Sep 17 00:00:00 2001
From: Stivvo <stivvo01@gmail.com>
Date: Thu, 11 Mar 2021 18:10:43 +0100
Subject: [PATCH 2/2] Faster setlayout(), floating=0 only selmon

- using setfloating() on each client is inefficient, since setfloating()
calls arrange(), which is going to be called at the end of setlayout()
anyway
- only set floating = 0 clients in the current monitor (bug fix)
---
 dwl.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/dwl.c b/dwl.c
index 0332ad45..7b8013f9 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1911,12 +1911,10 @@ setlayout(const Arg *arg)
 	if (!arg || !arg->v || arg->v != selmon->lt[selmon->sellt])
 		selmon->sellt ^= 1;
 	if (arg && arg->v) {
-		if (arg->v) { // setting a layout != floating
-			Client *c;
-			wl_list_for_each(c, &clients, link) {
-				setfloating(c, 0);
-			}
-		}
+		Client *c;
+		wl_list_for_each(c, &clients, link)
+			if (VISIBLEON(c, selmon))
+				c->isfloating = 0;
 		selmon->lt[selmon->sellt] = (Layout *)arg->v;
 	}
 	/* TODO change layout symbol? */
